### PR TITLE
added temperature schedules

### PR DESCRIPTION
**Pull request by M-R-Schaefer**
This PR adds temperature schedules to JaxMD.
The three schedules are Constant, PiecewiseLinear and the OscillatingRamp from IPSuite.
All combinations of thermostat and schedule work.

### CODE REVIEW

**Comment by M-R-Schaefer** [comment]
pre-commit.ci autofix

**Comment by M-R-Schaefer** [comment]
pre-commit.ci autofix

**Comment by M-R-Schaefer** [comment]
pre-commit.ci autofix

File: apax/cli/templates/md_config_minimal.yaml [hunk 1]
```diff
@@ -1,7 +1,9 @@
 ensemble:
   name: nvt
   dt: 0.5 # fs time step
-  temperature: <T> # K
+  temperature_schedule:
+    name: constant
+    T0: <T> # K
```
Comment (Tetracarbonylnickel) [inline apax/cli/templates/md_config_minimal.yaml:6]: I think we should document all the possible config inputs. Not only for the temp scheduler.

File: apax/config/md_config.py [hunk 2]
```diff
@@ -7,6 +7,64 @@
...
+class ConstantTempSchedule(BaseModel, extra="forbid"):
+    """
+    Attributes
+    ----------
+    temperature : PositiveFloat, default = 298.15
+        Temperature in Kelvin (K).
+    """
+
+    name: Literal["constant"] = "constant"
+    T0: PositiveFloat = 298.15  # K
```
Comment (Tetracarbonylnickel) [inline apax/config/md_config.py:19]: update docs

File: apax/config/md_config.py [hunk 3]
```diff
@@ -7,6 +7,64 @@
...
+
+    def get_schedule(self):
+        from apax.md.schedules import ConstantTSchedule
+
+        return ConstantTSchedule(self.T0)
+
+
+class PiecewiseLinearTempSchedule(ConstantTempSchedule, extra="forbid"):
+    name: Literal["piecewise"] = "piecewise"
```
Comment (Tetracarbonylnickel) [inline apax/config/md_config.py:28]: Add docs

File: apax/config/md_config.py [hunk 4]
```diff
@@ -7,6 +7,64 @@
...
+        )
+        return schedule
+
+
+class OscillatingRampTempSchedule(ConstantTempSchedule, extra="forbid"):
+    name: Literal["oscillating_ramp"] = "oscillating_ramp"
+    Tend: PositiveFloat
+    amplitude: PositiveFloat
+    num_oscillations: PositiveInt
+    total_steps: PositiveInt
```
Comment (Tetracarbonylnickel) [inline apax/config/md_config.py:48]: Add docs

**Review by Tetracarbonylnickel — CHANGES_REQUESTED** [review]

**Comment by Tetracarbonylnickel** [comment]
Looks good. After including docs I think we can merge.

File: apax/cli/templates/md_config_minimal.yaml (hunk 1 above)
Comment (M-R-Schaefer) [reply apax/cli/templates/md_config_minimal.yaml:6]: you mean with regards to the unit?

**Comment by M-R-Schaefer** [comment]
pre-commit.ci autofix

File: apax/cli/templates/md_config_minimal.yaml (hunk 1 above)
Comment (Tetracarbonylnickel) [reply apax/cli/templates/md_config_minimal.yaml:6]: No I mean in various parts of the config one can choose different temp schedules, ensemble types and so on. We only communicate one of these choices properly. I think we should think about a better way of communicating the different choices that can be made in the configuration script.

Comment (Tetracarbonylnickel) [reply apax/cli/templates/md_config_minimal.yaml:6]: This does not be addressed here in the PR I just recognised it.

**Review by Tetracarbonylnickel — APPROVED** [review]

**Pull request merged**